client/src/templates/Challenges/odin/Show.tsx

```diff
@@ -0,0 +1,431 @@
+  static displayName: string;
+  private _container: HTMLElement | null | undefined;
+
+  constructor(props: ShowOdinProps) {
+    super(props);
+    this.state = {
+      subtitles: '',
+      downloadURL: null,
+      selectedOption: null,
+      answer: 1,
+      showWrong: false,
+      assignmentsCompleted: 0,
+      allAssignmentsCompleted: false,
+      videoIsLoaded: false
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount(): void {
+    const {
+      challengeMounted,
+      data: {
+        challengeNode: {
+          challenge: { title, challengeType, helpCategory }
+        }
+      },
+      pageContext: { challengeMeta },
+      updateChallengeMeta
+    } = this.props;
+    updateChallengeMeta({
+      ...challengeMeta,
+      title,
+      challengeType,
+      helpCategory
+    });
+    challengeMounted(challengeMeta.id);
+    this._container?.focus();
+  }
+
+  componentDidUpdate(prevProps: ShowOdinProps): void {
+    const {
+      data: {
+        challengeNode: {
+          challenge: { title: prevTitle }
+        }
+      }
+    } = prevProps;
+    const {
+      challengeMounted,
+      data: {
+        challengeNode: {
+          challenge: { title: currentTitle, challengeType, helpCategory }
+        }
+      },
+      pageContext: { challengeMeta },
+      updateChallengeMeta
+    } = this.props;
+    if (prevTitle !== currentTitle) {
+      updateChallengeMeta({
+        ...challengeMeta,
+        title: currentTitle,
+        challengeType,
+        helpCategory
+      });
+      challengeMounted(challengeMeta.id);
+    }
+  }
+
+  handleSubmit(
+    solution: number,
+    openCompletionModal: () => void,
+    assignments: string[]
+  ) {
+    const hasAssignments = assignments[0] != '';
+    const completed = this.state.allAssignmentsCompleted;
+
+    if (solution - 1 == this.state.selectedOption && hasAssignments) {
+      this.setState({
+        showWrong: false
+      });
+
+      if (completed) {
+        openCompletionModal();
+      }
+    } else if (solution - 1 === this.state.selectedOption && !hasAssignments) {
+      this.setState({
+        showWrong: false
+      });
+      openCompletionModal();
+    } else if (solution - 1 !== this.state.selectedOption) {
+      this.setState({
+        showWrong: true
+      });
+    }
+  }
+
+  handleOptionChange = (
+    changeEvent: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    this.setState({
+      showWrong: false,
+      selectedOption: parseInt(changeEvent.target.value, 10)
+    });
+  };
+
+  handleAssignmentChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    assignments: number
+  ): void => {
+    const assCopy = this.state.assignmentsCompleted;
+    const completed = event.target.checked ? assCopy + 1 : assCopy - 1;
+    const allCompleted = assignments == completed;
+
+    this.setState({
+      assignmentsCompleted: completed,
+      allAssignmentsCompleted: allCompleted
```

Comment:
```suggestion
    totalAssignments: number
  ): void => {
    const assignmentsCompleted = event.target.checked
      ? this.state.assignmentsCompleted + 1
      : this.state.assignmentsCompleted - 1;
    const allAssignmentsCompleted = totalAssignments === assignmentsCompleted;

    this.setState({
      assignmentsCompleted,
      allAssignmentsCompleted
```
This reads a little better to me.